### PR TITLE
[21.05] python3Packages.scikit-learn: unbreak darwin

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
Backport of #124255

ZHF #122042 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build python3Packages.scikit-learn

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124414` at 52030b62 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.mlxtend</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.osmnx</li>
    <li>python38Packages.pygbm</li>
    <li>python38Packages.shap</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/1625dcd8f5b023b5946aa9e5a2b2f920">python38Packages.scikit-bio</a></li>
    <li><a href="https://gist.github.com/ea260b50f7821361ee99adae53eb7fb9">python39Packages.qiskit-ignis</a></li>
    <li><a href="https://gist.github.com/f91feda5be018f1e09f8d6118a59c90c">python39Packages.scikit-bio</a></li>
  </ul>
</details>
<details>
  <summary>22 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.graspologic</li>
    <li>python38Packages.hyppo</li>
    <li>python38Packages.nilearn</li>
    <li>python38Packages.pytorch-metric-learning</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.qiskit-ignis</li>
    <li>python38Packages.scikit-optimize</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>28 packages built successfully:</summary>
  <ul>
    <li>python38Packages.bayesian-optimization</li>
    <li>python38Packages.hdbscan</li>
    <li>python38Packages.hmmlearn</li>
    <li>python38Packages.imbalanced-learn</li>
    <li>python38Packages.kmapper</li>
    <li>python38Packages.librosa</li>
    <li>python38Packages.lightgbm</li>
    <li>python38Packages.mlrose</li>
    <li>python38Packages.persim</li>
    <li>python38Packages.ppscore</li>
    <li>python38Packages.pynndescent</li>
    <li>python38Packages.ripser</li>
    <li>python38Packages.scikitlearn (python38Packages.scikit-learn)</li>
    <li>python38Packages.seqeval</li>
    <li>python38Packages.word2vec</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.bayesian-optimization</li>
    <li>python39Packages.hmmlearn</li>
    <li>python39Packages.imbalanced-learn</li>
    <li>python39Packages.kmapper</li>
    <li>python39Packages.lightgbm</li>
    <li>python39Packages.mlrose</li>
    <li>python39Packages.persim</li>
    <li>python39Packages.ppscore</li>
    <li>python39Packages.ripser</li>
    <li>python39Packages.scikitlearn (python39Packages.scikit-learn)</li>
    <li>python39Packages.seqeval</li>
    <li>python39Packages.word2vec</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.
